src/test/java/team/serenity/logic/commands/studentinfo/UnflagAttCommandTest.java

```diff
@@ -0,0 +1,129 @@
+package team.serenity.logic.commands.studentinfo;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static team.serenity.commons.core.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+import static team.serenity.commons.core.Messages.MESSAGE_NOT_VIEWING_A_GROUP;
+import static team.serenity.commons.core.Messages.MESSAGE_NOT_VIEWING_A_LESSON;
+import static team.serenity.commons.core.Messages.MESSAGE_STUDENT_NOT_FOUND;
+import static team.serenity.testutil.Assert.assertThrows;
+import static team.serenity.testutil.TypicalIndexes.INDEX_FIRST;
+import static team.serenity.testutil.TypicalIndexes.INDEX_SECOND;
+import static team.serenity.testutil.TypicalStudent.AARON;
+import static team.serenity.testutil.TypicalStudent.JOHN;
+
+import org.junit.jupiter.api.Test;
+
+import team.serenity.commons.core.index.Index;
+import team.serenity.logic.commands.CommandResult;
+import team.serenity.logic.commands.exceptions.CommandException;
+import team.serenity.model.group.student.Student;
+
+class UnflagAttCommandTest {
+
+    @Test
+    public void constructor_nullParameter_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new UnflagAttCommand((Index) null));
+        assertThrows(NullPointerException.class, () -> new UnflagAttCommand((Student) null));
+    }
+
+    @Test
+    public void execute_unflagStudent_success() throws CommandException {
+        ModelStubWithStudentsFlagged modelStub = new ModelStubWithStudentsFlagged();
+        Student toUnflagAtt = new Student("Aaron Tan", "A0123456U");
```

Comment:
The current implementation of StudentBuilder fails the test case as the name of the Student does not correspond to name of any student in the lesson created by lesson builder, so I did not use it haha Do you think it will be better to change the default student name for Student Builder (not sure whether will fail test case elsewhere) or leave my current implementation as it is?

Reply:
oh I shall change the way the student is built and infuse in StudentBuilder like 'new StudentBuilder().withName("Aaron Tan").withId("A0123456U").build();' I think that should do the trick:)

Reply:
> The current implementation of StudentBuilder fails the test case as the name of the Student does not correspond to name of any student in the lesson created by lesson builder, so I did not use it haha Do you think it will be better to change the default student name for Student Builder (not sure whether will fail test case elsewhere) or leave my current implementation as it is?

I have decided to change the default student name to the correct one since it seems to make more sense that the default student build is a valid student. I fix some areas to pass the test cases too, they were pretty minor I realise, so that's good. I will proceed to use the default student build by `StudentBuilder`, thanks for the suggestion! It was a good point:)